spark/src/main/scala/org/apache/spark/sql/delta/DeltaParquetFileFormat.scala

```diff
@@ -469,6 +479,66 @@ object DeltaParquetFileFormat {
   /** Helper class that encapsulate an [[RowIndexFilterType]]. */
   case class DeletionVectorDescriptorWithFilterType(
       descriptor: DeletionVectorDescriptor,
-      filterType: RowIndexFilterType) {
+      filterType: RowIndexFilterType)
+
+  /**
+   * Translates the filter to use physical column names instead of logical column names.
+   * This is needed when the column mapping mode is set to `NameMapping` or `IdMapping`
+   * to match the requested schema that's passed to the [[ParquetFileFormat]].
+   */
+  private def translateFilterForColumnMapping(
+       filter: Filter,
+       physicalNameMap: Map[Seq[String], Seq[String]]): Option[Filter] = {
+    def getPhysicalName(attribute: String): String = {
+      import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.MultipartIdentifierHelper
+      physicalNameMap(parseColumnPath(attribute)).quoted
```

Comment:
For an extra layer of safety, you can use `Map.get` and drop the filter if there's no entry for that attribute in the map.


Also, this could be implemented as an extractor, e.g.:

```
    object PhysicalAttribute {
      import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.MultipartIdentifierHelper
      def unapply(attribute: String): Option[String] = {
        physicalNameMap.get(parseColumnPath(attribute)).map(_.quoted)
      }
    }

    filter match {
      case EqualTo(PhysicalAttribute(physicalAttr), value) =>
        Some(EqualTo(physicalAttr, value))
      case EqualNullSafe(PhysicalAttribute(physicalAttr), value) =>
        Some(EqualNullSafe(physicalAttr, value))
...
```
not that it makes a big difference, just a matter of personal preference